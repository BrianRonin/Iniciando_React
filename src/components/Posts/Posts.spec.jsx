const { render, screen } = require('@testing-library/react')
import { Posts } from '.'

const props = {
  posts: [
    {
      key: 1,
      conver: 'foto',
      title: 'Um titulo foda',
      body: 'Um body fodaaa',
      alt: 'Um alt fodaaaa',
    },
    {
      key: 2,
      conver: 'foto',
      title: 'Um titulo foda',
      body: 'Um body fodinha',
      alt: 'Um alt',
    },
    {
      key: 3,
      conver: 'foto',
      title: 'Um titulo foda',
      body: 'Um body fod√£o',
      alt: 'Um alt',
    },
  ],
}

describe('<Posts/>', () => {
  it('should render posts', () => {
    render(<Posts {...props} />)
    expect(
      screen.getAllByRole('heading', { name: /Um titulo foda/i }),
    ).toHaveLength(3) // eu espero que tenha selecionado 3
    // eu espero que tenha selecionado 3
    expect(screen.getAllByRole('img', { name: /Um alt/i })).toHaveLength(3)
    expect(screen.getAllByText(/Um body/i)).toHaveLength(3)
    expect(screen.getAllByText(/Um body/i)).toHaveLength(3)
  })
  it('should match snapshot', () => {
    const { container } = render(<Posts {...props} />)
    expect(container).toMatchSnapshot()
  })
})
