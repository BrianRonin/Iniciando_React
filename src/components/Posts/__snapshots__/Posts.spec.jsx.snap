// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts/> should match snapshot 1`] = `
<div>
  <div
    class="posts"
  >
    <div
      class="post"
    >
      <img
        alt="Um alt fodaaaa"
      />
      <div
        class="post-content"
      >
        <h1>
          Um titulo foda
        </h1>
        <p>
          Um body fodaaa
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="Um alt"
      />
      <div
        class="post-content"
      >
        <h1>
          Um titulo foda
        </h1>
        <p>
          Um body fodinha
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="Um alt"
      />
      <div
        class="post-content"
      >
        <h1>
          Um titulo foda
        </h1>
        <p>
          Um body fod√£o
        </p>
      </div>
    </div>
  </div>
</div>
`;
